django/db/backends/sqlite3/creation.py

```diff
@@ -101,3 +110,17 @@ def test_db_signature(self):
         else:
             sig.append(test_database_name)
         return tuple(sig)
+
+    def setup_worker_connection(self, _worker_id):
+        alias = self.connection.alias
+        worker_db = f'file:memorydb_{str(alias)}_{str(_worker_id)}?mode=memory&cache=shared'
+        sourcedb = sqlite3.connect(f'file:{str(alias)}_{str(_worker_id)}.sqlite3', uri=True)
+        second_db = sqlite3.connect(worker_db, uri=True)
+        sourcedb.backup(second_db)
+        sourcedb.close()
+        settings_dict = self.connection.settings_dict
+        settings_dict['NAME'] = worker_db
+        self.connection.settings_dict.update(settings_dict)
```

Comment:
I'm confused. Is there something incorrect here or can this just be:
```suggestion
        self.connection.settings_dict['NAME'] = worker_db
```

If the intent was to avoid mutating the original then do this (although I'm not sure it's required):

```python
        self.connection.settings_dict = {**self.connection.settings_dict, 'NAME': worker_db}
```

Reply:
Honestly not sure why I did it this way, will change it & see if something breaks